lnd_services.go

```diff
@@ -250,6 +260,24 @@ func NewLndServices(cfg *LndServicesConfig) (*GrpcLndServices, error) {
 
 	log.Infof("Using network %v", cfg.Network)
 
+	// If requested in the configuration, we now wait for lnd to fully sync
+	// to its chain backend. We do not add any timeout as it would be hard
+	// to determine a sane value. If the initial block download is still in
+	// progress, this could take hours.
+	if cfg.BlockUntilChainSynced {
+		log.Infof("Waiting for lnd to be fully synced to its chain " +
+			"backend, this might take a while")
+
+		err := services.waitForChainSync()
```

Comment:
Makes sense that we can't add a timeout, but is there perhaps still a need to allow caller to cancel at this stage? Eg, using this setting in loop, lnd isn't synced yet, we want to kill loopd and come back another day, our usual sigterm won't work. Or am I missing the point entirely with this option?

Reply:
No, you're absolutely right. Makes sense that we might want to cancel the wait on shutdown. Added an optional context that can be canceled.